packages/investor-foxy/src/api/api.ts

```diff
@@ -808,4 +809,52 @@ export class FoxyApi {
       releaseTime
     }
   }
+
+  async getRebaseHistory(input: BalanceInput) {
+    const { tokenContractAddress, userAddress } = input
+    this.verifyAddresses([tokenContractAddress])
+
+    const contract = new this.web3.eth.Contract(foxyAbi, tokenContractAddress)
+    const fromBlock = 14381454 // genesis rebase
+
+    const rebaseEvents = (
+      await contract.getPastEvents('LogRebase', {
+        fromBlock,
+        toBlock: 'latest'
+      })
+    ).filter((rebase) => rebase.returnValues.rebase !== '0')
+
+    const events: RebaseEvent[] = rebaseEvents.map((rebaseEvent) => {
+      const {
+        blockNumber,
+        returnValues: { epoch }
+      } = rebaseEvent
+      return {
+        blockNumber,
+        epoch
+      }
+    })
+
+    const rebaseChartData = await Promise.all(
+      events.map(async (event) => {
+        let balance
+        try {
+          balance = await contract.methods.balanceOf(userAddress).call(null, event.blockNumber)
+        } catch (e) {
+          throw new Error(`Failed to get balance of address at block ${e}`)
+        }
+
+        let timestamp
+        try {
+          timestamp = (await this.web3.eth.getBlock(event.blockNumber)).timestamp
+        } catch (e) {
+          throw new Error(`Failed to get timestamp of block ${e}`)
+        }
+
```

Comment:
do you want to use the const assignment IIFE pattern here?

Reply:
sure